test/unit/Makefile.am

```diff
@@ -182,6 +183,10 @@ mt_contention_LDFLAGS = $(PTHREAD_LIBS)
 mt_contention_CFLAGS = -I$(top_srcdir)/test/unit $(PTHREAD_CFLAGS)
 mt_contention_LDADD = $(LDADD) $(PTHREAD_CFLAGS)
 
+gettid_register_LDFLAGS = $(PTHREAD_LIBS)
+gettid_register_CFLAGS = -I$(top_srcdir)/test/unit $(PTHREAD_CFLAGS)
```

Comment:
You can safely remove `-I$(top_srcdir)/test/unit`.  This is only needed if the unit test includes the `pthread_barrier` shim header.